### DISCUSSION TITLE
Escaping problem when renaming jinja template from .htm to .jinja

### DISCUSSION THREAD

**Discussion by BobSquarePants** [pallets/flask#5332] [Q&A]
Hi everyone,

I don't know if the following is a unexpected behavior, so I wanted your advice before posting as an Issues.

My Jinja template file had extension `.htm` and everything was working smoothly.
But I change them to `.jinja` to directly have the right syntax color in my editor.

and weirdly changing the file extension mess it up when escaping characters more info --> https://github.com/brython-dev/brython/discussions/2313

> Flask  2.2.2
> Jinja2  3.1.2

Cheers

**Comment by bodik** [comment]
I'd say that this is expected and documented behavior https://flask.palletsprojects.com/en/3.0.x/templating/

> Jinja Setup
> Unless customized, Jinja2 is configured by Flask as follows:
> * autoescaping is enabled for all templates ending in .html, .htm, .xml, .xhtml, as well as .svg when using render_template().

**Comment by ThiefMaster** [comment]
you should configure your editor to applying jinja highlighting for `.html` files. at least in vscode it's really easy to do that
